core/src/cluster_info.rs

```diff
@@ -87,22 +88,26 @@ pub const GOSSIP_SLEEP_MILLIS: u64 = 100;
 /// The maximum size of a bloom filter
 pub const MAX_BLOOM_SIZE: usize = MAX_CRDS_OBJECT_SIZE;
 pub const MAX_CRDS_OBJECT_SIZE: usize = 928;
-/// The maximum size of a protocol payload
-const MAX_PROTOCOL_PAYLOAD_SIZE: u64 = PACKET_DATA_SIZE as u64 - MAX_PROTOCOL_HEADER_SIZE;
-/// The largest protocol header size
-const MAX_PROTOCOL_HEADER_SIZE: u64 = 214;
 /// A hard limit on incoming gossip messages
 /// Chosen to be able to handle 1Gbps of pure gossip traffic
 /// 128MB/PACKET_DATA_SIZE
 const MAX_GOSSIP_TRAFFIC: usize = 128_000_000 / PACKET_DATA_SIZE;
 
-/// Keep the number of snapshot hashes a node publishes under MAX_PROTOCOL_PAYLOAD_SIZE
+/// Keep the serialized size of SnapshotHashes/AccountsHashes a node publishes
+/// under ..._MAX_PAYLOAD_SIZE by limiting number of hashes in a CrdsValue.
+// TODO: Update this to 26 once payload sizes are upgraded across fleet.
 pub const MAX_SNAPSHOT_HASHES: usize = 16;
 /// Number of bytes in the randomly generated token sent with ping messages.
 const GOSSIP_PING_TOKEN_SIZE: usize = 32;
 const GOSSIP_PING_CACHE_CAPACITY: usize = 16384;
 const GOSSIP_PING_CACHE_TTL: Duration = Duration::from_secs(640);
 
+lazy_static::lazy_static! {
+    /// Maximum serialized sized of crds-values in a Protocol::PushMessage.
+    static ref PUSH_MESSAGE_MAX_PAYLOAD_SIZE: u64 = PACKET_DATA_SIZE as u64
```

Comment:
We've seen some bad performance with lazy_static in the past, best to do just a fixed constant and then verify that with a test.

Reply:
good idea, done.